h2o-core/src/test/java/water/UnknownHeartbeatTest.java

```diff
@@ -0,0 +1,28 @@
+package water;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class UnknownHeartbeatTest extends TestUtil{
+  @BeforeClass() public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  @Test
+  public void testIgnoreUnknownHeartBeat() {
+    HeartBeat hb = new HeartBeat();
+    hb._cloud_name_hash = 777;
+    hb._client = true;
+    hb._jar_md5 = H2O.SELF._heartbeat._jar_md5;
+
+    AutoBuffer ab = new AutoBuffer(H2O.SELF, H2O.MAX_PRIORITY);
```

Comment:
Did you try solution with ByteBuffer?

Reply:
Yup, tried. Didn't work. At the end, this works as expected.

Reply:
that's a pity, that would be nice way to communicate with h2o.
Do you have a reason/exception?

Reply:
We couldn't use purely just ByteBuffer as the `heartbeat.write` expects `AutoBuffer`.

So I filled the `AutoBuffer ab` first and then try to send it out using datagram channel, but the data hasn't probably been even sent out. I was debugging it for a while and then went for a quick solution with `AutoBuffer.close()`